tkltest/tkltest_unit.py

```diff
@@ -59,27 +61,53 @@ def main():
     tkltest_config = load_configuration(args, 'unit')
 
     configs = config_util.resolve_tkltest_configs(tkltest_config, args.command)
+    failed_modules = []
     for config in configs:
         logging_util.tkltest_status('{} tests for {} {} using config file {}.'.format(
             'Generating' if args.command == 'generate' else 'Executing',
             'module' if 'module_name' in config['general'] else 'app',
             config['general'].get('module_name', config['general']['app_name']),
             config['general'].get('config_file_path', args.config_file.name)))
 
-        unjar_paths = __unjar_path(config)
-        # process generate/execute commands
-        try:
-            if args.command == 'execute':
-                execute.process_execute_command(args, config)
-            elif args.command == 'generate':
-                 generate.process_generate_command(args, config)
-        finally:
-            for path in unjar_paths:
-                if os.path.isdir(path):
-                    shutil.rmtree(path)
+        if len(configs) == 1:
+            __process_command(args, config)
+        else:
+            # we can not deliver a TextIOWrapper as a part of an argument to a process,
+            # so we create a copy of args to and remove config_file from it
+            simple_args = SimpleNamespace(**vars(args))
+            del simple_args.config_file
+            process = Process(target=__process_command, args=(simple_args, config))
+            process.start()
+            process.join()
+            if process.exitcode:
+                failed_modules.append(config['general']['module_name'])
 
+    if failed_modules:
+        if len(failed_modules) == len(configs):
+            logging_util.tkltest_status('Failed to {} tests for all modules, Please see log for details'.format(args.command), error=True)
+            sys.exit(1)
+        else:
+            logging_util.tkltest_status('Warning: Failed to {} tests for the following modules: {}.'
+                                        ' Please see log for details'
+                                        .format(args.command, ', '.join(failed_modules)))
     if args.command == 'execute' and tkltest_config['execute']['combine_modules_coverage_reports']:
-        execute.merge_modules_coverage_reports(tkltest_config, configs)
+        execute.merge_modules_coverage_reports(tkltest_config, configs, failed_modules)
+
+
+
+def __process_command(args, config):
```

Comment:
Why implement a new process support instead of using one of the existing ones? (I think the first one - `run_command`, is sufficient)

1. https://github.com/konveyor/tackle-test-generator-cli/blob/fcb84ad22fd1248138bc8f6e917190d517733193/tkltest/util/command_util.py#L19
2. https://github.com/konveyor/tackle-test-generator-cli/blob/fcb84ad22fd1248138bc8f6e917190d517733193/tkltest/util/command_util.py#L43?

Reply:
I looked into it,
run_command() is for using an external executable.
 here I need that the new process will run another python method